go/api/base_client.go

```diff
@@ -198,6 +203,9 @@ func (client *baseClient) executeCommandWithRoute(
 	args []string,
 	route config.Route,
 ) (*C.struct_CommandResponse, error) {
+	client.mu.RLock()
+	defer client.mu.RUnlock()
```

Comment:
This `RLock` holds throughout the `executeCommandWithRoute` to make sure the corresponding `successCallback` or `failureCallback` is invoked before closing the `coreClient`.